jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditorDialog.kt

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import java.awt.event.ActionEvent
+import javax.swing.Action
+import javax.swing.JComponent
+import software.aws.toolkits.resources.message
+import java.time.temporal.ChronoUnit
+import java.util.Calendar
+import java.util.Date
+
+val relativeTimeUnit = mapOf(
+    message("cloudwatch.logs.time_minutes") to ChronoUnit.MINUTES,
+    message("cloudwatch.logs.time_hours") to ChronoUnit.HOURS,
+    message("cloudwatch.logs.time_days") to ChronoUnit.DAYS,
+    message("cloudwatch.logs.time_weeks") to ChronoUnit.WEEKS)
+
+class QueryEditorDialog(
+    private val project: Project,
+    private val lGroupName: String,
+    private val absoluteTimeSelected: Boolean = false
+
+) : DialogWrapper(project) {
+    constructor(project: Project, logGroupName: String) :
+        this(project = project, lGroupName = logGroupName)
+
+    private val view = QueryEditor(project)
+    private val queryingLogGroupApiCall = QueryingLogGroups(project)
+    private val action: OkAction = QueryLogGroupOkAction()
+    private val logGroupNames = mutableListOf(lGroupName)
```

Comment:
This one doesn't have to be mutable either

Reply:
logGroupNames will ultimately contain all log groups selected by the user on the form. So the other log groups will have to be added later